spec/scenarios/invoices/invoice_numbering_spec.rb

```diff
@@ -309,4 +315,435 @@
+          .to match_array(
+            %w[
+              ORG-1-202307-001
+              ORG-1-202307-004
+              ORG-1-202307-002
+              ORG-1-202307-003
+              ORG-1-DRAFT
+            ],
+          )
+      end
+
+      travel_to(time + 2.hours) do
+        draft_invoice_2 = customer_second.reload.invoices.draft.order(created_at: :asc).last
+
+        finalize_invoice(draft_invoice_2)
+
+        invoices = organization.reload.invoices.order(created_at: :desc)
+
+        expect(invoices.pluck(:number))
+          .to match_array(
+            %w[
+              ORG-1-202307-001
+              ORG-1-202307-004
+              ORG-1-202307-002
+              ORG-1-202307-003
+              ORG-1-202307-005
+            ],
+          )
+      end
+
+      # NOTE: August 19th: Bill subscription
+      travel_to(DateTime.new(2023, 8, 19, 12, 12)) do
+        Subscriptions::BillingService.call
+        perform_all_enqueued_jobs
+
+        invoices = organization.invoices.order(created_at: :desc).limit(3)
+        sequential_ids = invoices.pluck(:sequential_id)
+        organization_sequential_ids = invoices.pluck(:organization_sequential_id)
+        numbers = organization.reload.invoices.order(created_at: :desc).pluck(:number)
+
+        expect(sequential_ids).to match_array([3, nil, 2])
+        expect(organization_sequential_ids).to match_array([6, 0, 7])
+        expect(numbers)
+          .to match_array(
+            %w[
+              ORG-1-202307-001
+              ORG-1-202307-004
+              ORG-1-202307-002
+              ORG-1-202307-003
+              ORG-1-202307-005
+              ORG-1-202308-006
+              ORG-1-DRAFT
+              ORG-1-202308-007
+            ],
+          )
+      end
+
+      # NOTE: September 19th: Bill subscription
+      travel_to(DateTime.new(2023, 9, 19, 12, 12)) do
+        Subscriptions::BillingService.call
+        perform_all_enqueued_jobs
+
+        invoices = organization.invoices.order(created_at: :desc).limit(3)
+        sequential_ids = invoices.pluck(:sequential_id)
+        organization_sequential_ids = invoices.pluck(:organization_sequential_id)
+        numbers = organization.reload.invoices.order(created_at: :desc).pluck(:number)
+
+        expect(sequential_ids).to match_array([4, nil, 3])
+        expect(organization_sequential_ids).to match_array([8, 0, 9])
+        expect(numbers)
+          .to match_array(
+            %w[
+              ORG-1-202307-001
+              ORG-1-202307-004
+              ORG-1-202307-002
+              ORG-1-202307-003
+              ORG-1-202307-005
+              ORG-1-202308-006
+              ORG-1-DRAFT
+              ORG-1-202308-007
+              ORG-1-202309-008
+              ORG-1-DRAFT
+              ORG-1-202309-009
+            ],
+          )
+      end
+
+      travel_to(DateTime.new(2023, 9, 20, 12, 12)) do
+        draft_invoice_1 = customer_second.reload.invoices.draft.order(created_at: :asc).first
```

Comment:
Naming/VariableNumber: Use normalcase for variable numbers. (https://rubystyle.guide#snake-case-symbols-methods-vars-with-numbers)